### PR TITLE
Fix uninitialized variable

### PR DESCRIPTION
**Pull request by kronbichler**
I saw this uninitialized variable when running a relaxation preconditioner with valgrind, and I believe it will also fix the error reported here:
https://cdash.dealii.org/viewTest.php?onlyfailed&buildid=2516

Related to #16747.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]

**Comment by kronbichler** [comment]
With this change, the test is now failing in the first three lines,
```
2,4c2,4
< DEAL::OK!
< DEAL::OK!
< DEAL::OK!
---
> DEAL::ERROR!
> DEAL::ERROR!
> DEAL::ERROR!
```
This corresponds to the relaxation parameter `0.0`, i.e., automatic determination of eigenvalues, running for 1, 2, 3 sweeps. However, the test is designed in such a way that it is really hard to see where exactly we are failing now. @peterrum would you mind looking into the tests as to where they fail? I think this just succeeded by chance before, depending on the spurious value of uninitialized variable. In any case, it would also be desirable if the test could print more precisely where something fails, rather than just accumulate errors in a spaghetti manner over 4 different metrics and 5 different test cases within each variant without printing specific information here:
https://github.com/dealii/dealii/blob/3ccc91fdd08c6f46fa3b5c87b61b99526bdcff43/tests/lac/precondition_relaxation_01.cc#L346-L360

**Comment by peterrum** [comment]
I will do.

**Comment by kronbichler** [comment]
Thanks for the help!

**Comment by peterrum** [comment]
@kronbichler I have fixed the test and added some asserts.

**Comment by kronbichler** [comment]
Thank you @peterrum, this looks much better now!

File: tests/lac/precondition_relaxation_01.output [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+DEAL::OK! 15.3680 24.2003 15.3680 24.2003 
+DEAL::OK! 5.62500 14.2615 5.62500 14.2615 
+DEAL::OK! 9.85490 18.6445 9.85490 18.6445 
+DEAL::OK! 14.3212 23.1484 14.3212 23.1484 
+DEAL::OK! 5.06250 13.7224 5.06250 13.7224 
+DEAL::OK! 8.97855 17.7551 8.97855 17.7551 
+DEAL::OK! 12.9800 21.7988 12.9800 21.7988 
+DEAL::OK! 6.18750 14.8024 6.18750 14.8024 
+DEAL::OK! 10.7121 19.5161 10.7121 19.5161 
+DEAL::OK! 15.6708 24.5042 15.6708 24.5042
```
Comment (tamiko) [inline tests/lac/precondition_relaxation_01.output:13]: Let's make this output more than 10 digits. Otherwise, numdiff might trigger accidentally.

Comment (kronbichler) [reply tests/lac/precondition_relaxation_01.output]: Good idea, I adjusted the test and its output.

**Pull request merged**